Bindings/Packages/com.virtualmaker.bindings/Runtime/Bindings.PlayerPrefs.cs

```diff
@@ -0,0 +1,97 @@
+        public void BindPlayerPref(string key, Property<int> prop, bool getFromPref, int? defaultValue = null)
+        {
+            if (getFromPref)
+            {
+                prop.Value = PlayerPrefs.GetInt(key, defaultValue ?? prop.Value);
+            }
+            BindPlayerPref(key, prop);
+        }
+
+        /// <summary> Binds player pref to property. </summary>
+        /// <param name="key"> Key used by <c>PlayerPrefs</c> to set the value. </param>
+        /// <param name="prop"> <c>IProperty</c> that sets the value. </param>
+        public void BindPlayerPref(string key, IProperty<int> prop)
+        {
+            Bind(prop, value => PlayerPrefs.SetInt(key, value));
+            SavePrefsOnReset();
+        }
+
+        /// <summary> Binds player pref to property. </summary>
+        /// <param name="key"> Key used by <c>PlayerPrefs</c> to get and set the value. </param>
+        /// <param name="prop"> Property that sets the value. </param>
+        /// <param name="getFromPref"> Whether to try setting <c>prop.Value</c> to the value already inside the <c>PlayerPrefs</c>. </param>
+        /// <param name="defaultValue"> Value to set <c>prop.Value</c> to if <c>getFromPref</c> is <c>true</c> but there isn't a value already inside the <c>PlayerPref</c>. </param>
+        public void BindPlayerPref(string key, Property<string> prop, bool getFromPref, string defaultValue)
+        {
+            if (getFromPref)
+            {
+                prop.Value = PlayerPrefs.GetString(key, defaultValue);
+            }
+            BindPlayerPref(key, prop);
+        }
+
+        /// <summary> Binds player pref to property. </summary>
+        /// <param name="key"> Key used by <c>PlayerPrefs</c> to get and set the value. </param>
+        /// <param name="prop"> Property that sets the value. </param>
+        /// <param name="getFromPref"> Whether to try setting <c>prop.Value</c> to the value already inside the <c>PlayerPrefs</c>. </param>
+        public void BindPlayerPref(string key, Property<string> prop, bool getFromPref)
+        {
+            if (getFromPref)
+            {
+                prop.Value = PlayerPrefs.GetString(key, prop.Value);
+            }
+            BindPlayerPref(key, prop);
+        }
+
+        /// <summary> Binds player pref to property. </summary>
+        /// <param name="key"> Key used by <c>PlayerPrefs</c> to set the value. </param>
+        /// <param name="prop"> <c>IProperty</c> that sets the value. </param>
+        public void BindPlayerPref(string key, IProperty<string> prop)
+        {
+            Bind(prop, value => PlayerPrefs.SetString(key, value));
+            SavePrefsOnReset();
+        }
+
+        private void SavePrefsOnReset()
+        {
+            if (!_unsubscribe.Contains(PlayerPrefs.Save))
+            {
+                AddUnsubscriber(PlayerPrefs.Save);
+            }
+        }
+    }
```

Comment:
What is the benefit of this if we set the player pref key value?

Reply:
Hmm, I figured it has a purpose but it says in the docs

> Saves all modified preferences.
> 
> Unity saves preferences automatically during OnApplicationQuit(). On the Universal Windows Platform, Unity writes preferences during application suspend. For information on the storage location, see [PlayerPrefs](https://docs.unity3d.com/ScriptReference/PlayerPrefs.html).
> 
> **Note:** Since writing the PlayerPrefs can cause hiccups, it is recommended to not call this function during gameplay.

So I'm lost on the purpose, unless for maybe if it's being changed in editor?

Reply:
Sounds good to me. I thought it usually saved as soon as the value was written

Reply:
saving here seems arbitrary to me. Reset is usually called when the component is disabled, so you're saving at that point?

I think if there's a good place to checkpoint, the application would do it itself by calling save.